src/HealthChecks.ClickHouse/ClickHouseHealthCheck.cs

```diff
@@ -0,0 +1,43 @@
+using System.Diagnostics;
+using ClickHouse.Client.ADO;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.ClickHouse;
+
+/// <summary>
+/// A health check for ClickHouse databases.
+/// </summary>
+public class ClickHouseHealthCheck : IHealthCheck
+{
+    private readonly ClickHouseHealthCheckOptions _options;
+
+    public ClickHouseHealthCheck(ClickHouseHealthCheckOptions options)
+    {
+        Debug.Assert(options.ConnectionString is not null || options.ConnectionFactory is not null);
+        Guard.ThrowIfNull(options.CommandText, true);
+        _options = options;
+    }
+
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            await using var connection = _options.ConnectionFactory?.Invoke() ?? new ClickHouseConnection(_options.ConnectionString);

```

Comment:
What are the recommendations for using `ClickHouseConnection`? Namely:
- is it thread safe?
- does it have a built-in pool of connections? (when we execute `new ClickHouseConnection(_options.ConnectionString)`, does it always create a new coonection?)

I need to get a better understanding of how it should be used before I approve the shape of the public API for this DB.

Reply:
@smbecker thank you for checking everything and contributing a PR!

Based on what you wrote and https://github.com/DarkWanderer/ClickHouse.Client/wiki/Connection-lifetime-&-pooling#recommendations:

> If an application operates on large volumes of transactions and requires to create/destroy ClickHouseConnection objects often, it is recommended to use IHttpClientFactory or a static instance of HttpClient to manage connections

I wonder if we should do the following:

1. Recommend the users to use `IHttpClientFactory` in the README.md and show an example that registers `ClickHouseConnection` factory method along with the `IHttpClientFactory`  (people just copy this code most of the time, this is why I would like it to be the "right" way of doing things)
2. Have a single extension method that takes a mandatory `Func<IServiceProvider, ClickHouseConnection> connectionFactory` argument. It allows the user to obtain all kinds of configuration options from the service provider and forces them to create `ClickHouseConnection` on their own.

```cs
public static IHealthChecksBuilder AddClickHouse(
    this IHealthChecksBuilder builder,
    Func<IServiceProvider, ClickHouseConnection> connectionFactory,
    string healthQuery = HEALTH_QUERY,
    Action<ClickHouseConnection>? configure = null,
    string? name = default,
    HealthStatus? failureStatus = default,
    IEnumerable<string>? tags = default,
    TimeSpan? timeout = default)
```

@smbecker what do you think?